File: day-02/workspace/java-oop/src/main/java/com/targetindia/programs/InterfaceDemo.java
package com.targetindia.programs;

import com.targetindia.model.Circle;
import com.targetindia.model.GeometricShape;
import com.targetindia.model.Triangle;

public class InterfaceDemo {

    static void processGeometricShape(GeometricShape shape){
        System.out.printf("Name of the shape is %s\n", shape.getShapeName());
        System.out.printf("Area of this %s is %f Sq.units\n", shape.getShapeName(), shape.calculateArea());
    }

    public static void main(String[] args) {
        Circle c1 = new Circle(2.34);
        Triangle t1 = new Triangle(2.3, 4.5);

        processGeometricShape(c1);
        processGeometricShape(t1);
    }
}

File: day-13/workspace/kotlin-interfaces/src/main/kotlin/com/targetindia/programs/InterfaceDemo.kt
package com.targetindia.programs

import com.targetindia.model.Circle
import com.targetindia.model.IGeometricShape
import com.targetindia.model.Triangle

// polymorphic method

fun processGeometricShape(shape: IGeometricShape){
    println("Name of the shape given is ${shape.getName()}")
    println("It\'s area is ${shape.getArea()}")
}

fun main(){
    val c1 = Circle(12.34);
    val t1 = Triangle(12.34, 56.78)

    processGeometricShape(c1)
    processGeometricShape(t1)

}